src/emulate.c

```diff
@@ -758,6 +765,34 @@ static uint32_t csr_csrrc(struct riscv_t *rv, uint32_t csr, uint32_t val)
     return out;
 }
 
+/* Performance counter API
```

Comment:
Why would you add pseudo-instructions here? We shall always implement the necessary functionalities first.

Reply:
Your right! I should add 
- [] 3. Implementing the incrementation of CSR_CYCLEH, CSR_TIME and CSR_INSTRT in rv_step()`
before providing this API.